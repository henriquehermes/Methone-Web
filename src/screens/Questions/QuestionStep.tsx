import React from "react";

import {
	Container,
	Title,
	Description,
	Header,
	QuestionNumber,
	Body,
	Box,
	Footer,
	Button,
	Question,
	Like,
	Dislike,
	StepLevel,
	Content,
	QuestionLevel,
} from "./styles";
import { QUESTIONS_STRINGS } from "../../language";
import Images from "../../assets/images";
import { TYPE_ANSWER } from "../../utils/Constants";

interface Data {
	category: string;
	subcategory?: string;
	type: string;
	difficulty: string;
	question: string;
	correct_answer: string;
	incorrect_answers: Array<string>;
}

interface Props {
	data: Data;
	nextStep: (action: string) => void;
	position: number;
}

const QuestionStep: React.FC<Props> = ({ data, nextStep, position }) => {
	return (
		<Container>
			<Header>
				<Title>{data.category}</Title>
				<Description>{data.subcategory}</Description>
			</Header>
			<Body>
				<Box>
					<StepLevel />
					<Content>
						<QuestionLevel>{data.difficulty}</QuestionLevel>
						<Question>{data.question}</Question>
					</Content>
				</Box>
				<QuestionNumber>
					{QUESTIONS_STRINGS.description}
					{position}
					{QUESTIONS_STRINGS.of}10
				</QuestionNumber>
			</Body>
			<Footer>
				<Button onClick={() => nextStep(TYPE_ANSWER.correct)}>
					<Like src={Images.likeIcon} />
				</Button>
				<Button onClick={() => nextStep(TYPE_ANSWER.incorrect)}>
					<Dislike src={Images.dislikeIcon} />
				</Button>
			</Footer>
		</Container>
	);
};

export default QuestionStep;
