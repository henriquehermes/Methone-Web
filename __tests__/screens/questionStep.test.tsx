import React from "react";
import renderer from "react-test-renderer";

import QuestionStep from "../../src/screens/Questions/QuestionStep";

const questions = {
	category: "Entertainment",
	subcategory: "Video Games",
	type: "boolean",
	difficulty: "hard",
	question: "Unturned originally started as a Roblox game.",
	correct_answer: "True",
	incorrect_answers: ["False"],
};

const newWrapper = () => (
	<QuestionStep data={questions} nextStep={jest.fn()} position={5} />
);

describe("screens - QuestionStep", () => {
	it("renders correctly", () => {
		const tree = renderer.create(newWrapper()).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
