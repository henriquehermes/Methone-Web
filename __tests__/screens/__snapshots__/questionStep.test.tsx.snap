// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screens - QuestionStep renders correctly 1`] = `
<div
  className="sc-AxjAm iPzjKo"
>
  <div
    className="sc-fznyAO EnZLe"
  >
    <div
      className="sc-fzpans iQTnjG"
    />
    <div
      className="sc-AxirZ dedhls"
    >
      <h2
        className="sc-AxiKw bbkWUD"
      >
        Entertainment
      </h2>
      <h1
        className="sc-AxhCb dXWuoR"
      >
        Video Games
      </h1>
    </div>
    <div
      className="sc-AxgMl jdXybX"
    >
      <h3
        className="sc-fzplWN cFPNxj"
      >
        hard
      </h3>
      <h2
        className="sc-AxmLO ksgqow"
      >
        Unturned originally started as a Roblox game.
      </h2>
    </div>
    <h3
      className="sc-AxhUy crKbCd"
    >
      Question 
      5
       of 
      10
    </h3>
    <div
      className="sc-AxheI eSPeMa"
    >
      <button
        className="sc-Axmtr kVUCYN"
        onClick={[Function]}
      >
        <img
          className="sc-fzozJi kIPTnD"
          src={
            Object {
              "process": [Function],
            }
          }
        />
      </button>
      <button
        className="sc-Axmtr kVUCYN"
        onClick={[Function]}
      >
        <img
          className="sc-fzoLsD bPkcit"
          src={
            Object {
              "process": [Function],
            }
          }
        />
      </button>
    </div>
  </div>
</div>
`;
